doc/optimizers.md

```diff
@@ -986,6 +986,62 @@ optimizer.Optimize(f, coordinates);
  * [Limited-memory BFGS in Wikipedia](https://en.wikipedia.org/wiki/Limited-memory_BFGS)
  * [Differentiable functions](#differentiable-functions)
 
+## Lookahead
+
+*An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+Lookahead is a stochastic gradient based optimization method which chooses a
+search direction by looking ahead at the sequence of "fast weights" generated
+by another optimizer.
```

Comment:
`Lookahead` seems different from our other optimizers in that it requires an inner optimizer type.  I think we could either point that out in the documentation here, meaning that the constructors documentation should be adapted to `Lookahead<BaseOptimizerType>(...)`, or, we could make `Adam` the default and also provide constructor overloads that don't need `baseOptimizer` so a user can construct it without needing to think about the base optimizer.  What do you think?